From 69f118c02988c28eb5c2bbae08e26759c99bb4b5 Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Thu, 29 Jan 2026 09:35:46 +0100
Subject: [PATCH 29/32] MdeModulePkg/SmbiosMeasurementDxe: fix gcc 16 warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c: In function ‘GetSmbiosStringById’:
MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c:221:10: error: variable ‘Size’ set but not used [-Werror=unused-but-set-variable=]
  221 |   UINTN  Size;
      |          ^~~~

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
---
 .../Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c     | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c b/MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c
index 9c7d3906be80..5388fe4c2876 100644
--- a/MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c
+++ b/MdeModulePkg/Universal/SmbiosMeasurementDxe/SmbiosMeasurementDxe.c
@@ -218,14 +218,12 @@ GetSmbiosStringById (
   OUT  UINTN                    *StringLen
   )
 {
-  UINTN  Size;
   UINTN  StrLen;
   CHAR8  *CharInStr;
   UINTN  StringsNumber;
   CHAR8  *String;
 
   CharInStr     = (CHAR8 *)Head + Head->Length;
-  Size          = Head->Length;
   StringsNumber = 0;
   StrLen        = 0;
   //
@@ -234,7 +232,6 @@ GetSmbiosStringById (
   String = NULL;
   while (*CharInStr != 0 || *(CharInStr+1) != 0) {
     if (*CharInStr == 0) {
-      Size += 1;
       CharInStr++;
     }
 
@@ -256,7 +253,6 @@ GetSmbiosStringById (
     // forward the pointer
     //
     CharInStr     += StrLen;
-    Size          += StrLen;
     StringsNumber += 1;
     if (StringsNumber == StringId) {
       break;
-- 
2.52.0

